pkg/apb/executor.go

```diff
@@ -80,27 +77,31 @@ func ExecuteApb(
 	if err != nil {
 		return executionContext, err
 	}
-	if len(secrets) > 0 {
-		executionContext.Namespace = clusterConfig.Namespace
-		executionContext.Targets = append(executionContext.Targets, context.Namespace)
-	} else {
-		// Using a new UUID is sane, because it will have some gurantee
-		// of uniquenes and will meet DNS name requirements.
-		executionContext.Namespace = uuid.New()
-		executionContext.Targets = append(executionContext.Targets, context.Namespace)
-		// Create namespace.
-		namespace := v1.Namespace{
-			ObjectMeta: metav1.ObjectMeta{
-				Labels: map[string]string{transientNameSpaceKey: spec.FQName},
-				Name:   executionContext.Namespace,
-			},
-		}
-		_, err := k8scli.CoreV1().Namespaces().Create(&namespace)
-		if err != nil {
-			return executionContext, err
-		}
+
+	labels := map[string]string{
+		"apb-fqname": spec.FQName,
+		"apb-action": action,
 	}
+
+	executionContext.Targets = append(executionContext.Targets, context.Namespace)
+	// Create namespace.
+	namespace := v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:       labels,
+			GenerateName: fmt.Sprintf("%s-%.4s-", spec.FQName, action),
```

Comment:
What's the upper limit on the size of this field? 64 chars? We should guarantee we're under it.

Reply:
I can make this change if you like, but my understanding is that GenerateName handles this for you:

Names generated by the system may be requested using metadata.generateName. GenerateName indicates that the name should be made unique by the server prior to persisting it. A non-empty value for the field indicates the name will be made unique (and the name returned to the client will be different than the name passed). The value of this field will be combined with a unique suffix on the server if the Name field has not been provided. The provided value must be valid within the rules for Name, and **may be truncated by the length of the suffix required to make the value unique on the server**. If this field is specified, and Name is not present, the server will NOT return a 409 if the generated name exists - instead, it will either return 201 Created or 504 with Reason ServerTimeout indicating a unique name could not be found in the time allotted, and the client should retry (optionally after the time indicated in the Retry-After header).

From https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#idempotency

Reply:
No change needed, that looks good! Thank you for the link :+1: